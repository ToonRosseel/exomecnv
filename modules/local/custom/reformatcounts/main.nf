//REFORMAT SAMTOOLS BEDCOV FILES (add header and remove 5th column)
process CUSTOM_REFORMATCOUNTS {
    tag "$meta.id"

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/coreutils:9.3' :
        'biocontainers/coreutils:9.3' }"
    conda "${moduleDir}/environment.yml"

    input:
    tuple val(meta), path(tsv)

    output:
    tuple val(meta), path("*.txt"), emit:header
    path "versions.yml", emit:versions

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def unzipped_tsv = tsv.toString().replaceAll('\\.gz$', '')
    """
    # Only decompress if the file ends with .gz
    if [[ "$tsv" == *.gz ]]; then
        gzip -d -f $tsv
    fi

    { echo -e "chromosome\\tstart\\tend\\texon\\t${meta.id}"; \
    cut --complement -f6 $unzipped_tsv ; } > "${prefix}.txt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        echo: \$(echo --version | sed '1!d; s/echo (GNU coreutils) //')
        cut: \$(cut --version | sed '1!d; s/cut (GNU coreutils) //')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        echo: \$(echo --version | sed '1!d; s/echo (GNU coreutils) //')
        cut: \$(cp --version | sed '1!d; s/cut (GNU coreutils) //')
    END_VERSIONS
    """
}
