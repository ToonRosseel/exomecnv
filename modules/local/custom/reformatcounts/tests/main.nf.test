nextflow_process {

    name "Test Process CUSTOM_REFORMATCOUNTS"
    script "../main.nf"
    process "CUSTOM_REFORMATCOUNTS"

    tag "modules"
    tag "modules_local"
    tag "reformatcounts"

    test("homo_sapiens - reformat") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', batch: 'prep_M' , family: 'Fam1', samples:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], families:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"], chr:'autosomal'],
                    file(params.counts1_to_reformat, checkIfExists:true)
                ])
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

    test("homo_sapiens - reformat - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', batch: 'prep_M' , family: 'Fam1', samples:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], families:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"], chr:'autosomal'],
                    file(params.counts1_to_reformat, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
