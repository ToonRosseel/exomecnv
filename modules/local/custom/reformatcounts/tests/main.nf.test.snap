{
    "homo_sapiens - reformat": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "batch": "prep_M",
                            "family": "Fam1",
                            "samples": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "families": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ],
                            "chr": "autosomal"
                        },
                        "Sample1.null.txt:md5,15e7b5fef592e6c1a0cd6956492f4413"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1fc282562f70ae81332ac5efb8314e49"
                ],
                "header": [
                    [
                        {
                            "id": "Sample1",
                            "batch": "prep_M",
                            "family": "Fam1",
                            "samples": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "families": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ],
                            "chr": "autosomal"
                        },
                        "Sample1.null.txt:md5,15e7b5fef592e6c1a0cd6956492f4413"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1fc282562f70ae81332ac5efb8314e49"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-08-25T13:44:37.980772002"
    },
    "homo_sapiens - reformat - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "batch": "prep_M",
                            "family": "Fam1",
                            "samples": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "families": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ],
                            "chr": "autosomal"
                        },
                        "Sample1.null.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3e09d87978796a478b6bd8edd09c4985"
                ],
                "header": [
                    [
                        {
                            "id": "Sample1",
                            "batch": "prep_M",
                            "family": "Fam1",
                            "samples": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "families": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ],
                            "chr": "autosomal"
                        },
                        "Sample1.null.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3e09d87978796a478b6bd8edd09c4985"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-07T13:25:03.940928"
    }
}